auth0/v3/authentication/base.py

```diff
@@ -42,16 +42,15 @@ def __init__(self, domain, telemetry=True):
 
     def post(self, url, data=None, headers=None):
         request_headers = self.base_headers.copy()
-        request_headers.update(headers)
-        response = requests.post(url=url, data=json.dumps(data),
-                                 headers=request_headers)
+        request_headers.update(headers or {})
+        response = requests.post(url=url, json=data, headers=request_headers)
         return self._process_response(response)
 
     def get(self, url, params=None, headers=None):
         request_headers = self.base_headers.copy()
-        request_headers.update(headers)
+        request_headers.update(headers or {})
         response = requests.get(url=url, params=params, headers=request_headers)
-        return response.text
+        return self._process_response(response)
```

Comment:
Some response formats where not taken into account. This was the only method not calling `_process_response`. Now it handles text responses (what was there before), JSON and empty body responses.